src/utils.lisp

```diff
@@ -10,11 +10,12 @@
 
 (defparameter *build* nil)
 
-(defun pad-list (list pad length)
-  (if (>= (length list) length)
-      list
-      (append (make-list (- length (length list)) :initial-element pad)
-              list)))
+(defun pad-list (list pad length
+                 &aux (pad-length (- length (length list))))
```

Comment:
nit/personal prefs: I don't think &aux makes things more readable, so I'll change it to a let in a followup

Reply:
Fine by me.